mirage/config.js

```diff
@@ -165,6 +165,29 @@ export default function () {
     );
   });
 
+  this.get('/coupons/:coupon/', function (schema, { params }) {
+    if (params.coupon === 'percentOff') {
+      return {
+        '@type': 'coupon',
+        '@representation': 'standard',
+        'id': '10_BUCKS_OFF',
+        'name': '10 bucks off!',
+        'percent_off': 10,
+        'amount_off': null,
+        'valid': true
+      };
+    }
+    return {
+      '@type': 'coupon',
+      '@representation': 'standard',
+      'id': '10_BUCKS_OFF',
+      'name': '10 bucks off!',
+      'percent_off': null,
+      'amount_off': 1000,
+      'valid': true
+    };
```

Comment:
It could be better to add a Mirage model and factory and generate as many coupons in tests as needed.